### PR TITLE
Fixed many clang-tidy warnings (from clang-analyzer-*)

### PR DESCRIPTION
**Pull request by FlorianReimold**

### CODE REVIEW

File: app/rec/rec_gui/src/qecalrec.cpp [hunk 1]
```diff
@@ -145,6 +145,9 @@ std::map<std::string, eCAL::rec_server::ClientConfig> QEcalRec::enabledRecClient
 bool QEcalRec::setHostFilter(const std::string& hostname, const std::set<std::string>& host_filter, bool omit_dialogs)
 {
   bool success = rec_server_->SetHostFilter(hostname, host_filter);
+  if (!success)
```
Comment (FlorianReimold) [inline app/rec/rec_gui/src/qecalrec.cpp:148]: I guess this is not the proper way to solve this. I rather think that either line is redundand:

```
  bool success = rec_server_->SetHostFilter(hostname, host_filter);
  success = executeBlockingMethod<bool>([this, &hostname, &host_filter]() { return rec_server_->SetHostFilter(hostname, host_filter); }, widgetOf(sender()), omit_dialogs);
```

They appear to do the same.

File: ecal/core/src/readwrite/ecal_reader.cpp [hunk 2]
```diff
@@ -765,7 +765,7 @@ namespace eCAL
       }
 
       // check for clock difference
-      if ((m_clock - m_clock_old) > 0)
+      else if ((m_clock - m_clock_old) > 0)
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader.cpp:768]: Honestly I like the previous version a little bit better.

File: ecal/core/src/readwrite/ecal_reader_udp_mc.cpp [hunk 3]
```diff
@@ -25,5 +25,4 @@
 
 namespace eCAL
 {
-  template<> std::shared_ptr<CMulticastLayer> CReaderLayer<CMulticastLayer>::layer(nullptr);
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader_udp_mc.cpp:28]: Why was this in here after all? Was this some residue from ancient times that now can be removed safely?

File: ecal/core/src/readwrite/ecal_reader_tcp.cpp [hunk 4]
```diff
@@ -37,7 +37,6 @@
 
 namespace eCAL
 {
-  template<> std::shared_ptr<CTCPReaderLayer> CReaderLayer<CTCPReaderLayer>::layer(nullptr);
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader_tcp.cpp:40]: Why was this in here after all? Was this some residue from ancient times that now can be removed safely?

File: ecal/core/src/readwrite/ecal_reader_shm.cpp [hunk 5]
```diff
@@ -39,7 +39,6 @@
 
 namespace eCAL
 {
-  template<> std::shared_ptr<CSHMLayer> CReaderLayer<eCAL::CSHMLayer>::layer(nullptr);
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader_shm.cpp:42]: Why was this in here after all? Was this some residue from ancient times that now can be removed safely?

File: ecal/core/src/readwrite/ecal_reader_inproc.cpp [hunk 6]
```diff
@@ -25,7 +25,6 @@
 
 namespace eCAL
 {
-  template<> std::shared_ptr<CInProcLayer> CReaderLayer<CInProcLayer>::layer(nullptr);
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader_inproc.cpp:28]: Why was this in here after all? Was this some residue from ancient times that now can be removed safely?

File: ecal/core/src/readwrite/ecal_reader_iceoryx.cpp [hunk 7]
```diff
@@ -34,7 +34,6 @@
 
 namespace eCAL
 {
-  template<> std::shared_ptr<CSHMLayer> CReaderLayer<CSHMLayer>::layer(nullptr);
```
Comment (FlorianReimold) [inline ecal/core/src/readwrite/ecal_reader_iceoryx.cpp:37]: Why was this in here after all? Was this some residue from ancient times that now can be removed safely?

File: ecal/core/src/ecal_process.cpp [hunk 8]
```diff
@@ -264,7 +264,8 @@ namespace eCAL
       sstream << "------------------------- PUBLISHER LAYER DEFAULTS ---------------"       << std::endl;
       sstream << "Layer Mode INPROC        : " << LayerMode(Config::GetPublisherInprocMode())  << std::endl;
       auto zero_copy = Config::IsMemfileZerocopyEnabled();
-      if (zero_copy)
+
+      if (true == zero_copy)
```
Comment (FlorianReimold) [inline ecal/core/src/ecal_process.cpp:268]: I don't understand this "improvement". Shouldn't you like always just use the boolean directly and not compare it to `true` or `false` manually?

**Review by FlorianReimold — COMMENTED** [review]
I Looked over all files and added some inline comments where I felt that an additional pair of eyes is needed

File: ecal/core/src/readwrite/ecal_reader_udp_mc.cpp (hunk 3 above)
Comment (rex-schilasky) [reply ecal/core/src/readwrite/ecal_reader_udp_mc.cpp:28]: This can not be removed because it is a the static global layer instance.

Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader_udp_mc.cpp:28]: We tracked it back and this is fine

File: ecal/core/src/readwrite/ecal_reader_tcp.cpp (hunk 4 above)
Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader_tcp.cpp:40]: We tracked it back and this is fine

File: ecal/core/src/readwrite/ecal_reader_shm.cpp (hunk 5 above)
Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader_shm.cpp:42]: We tracked it back and this is fine

File: ecal/core/src/readwrite/ecal_reader_iceoryx.cpp (hunk 7 above)
Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader_iceoryx.cpp:37]: We tracked it back and this is fine

File: ecal/core/src/readwrite/ecal_reader_inproc.cpp (hunk 6 above)
Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader_inproc.cpp:28]: We tracked it back and this is fine

File: ecal/core/src/readwrite/ecal_reader.cpp (hunk 2 above)
Comment (FlorianReimold) [reply ecal/core/src/readwrite/ecal_reader.cpp:768]: I will leave the else if to prevent the clang-tidy warning.

**Review by rex-schilasky — CHANGES_REQUESTED** [review]

File: ecal/core/include/ecal/ecal_qos.h [hunk 9]
```diff
@@ -75,7 +77,7 @@ namespace eCAL
       }
       eQOSPolicy_HistoryKind  history_kind;              //!< qos history kind mode
       int                     history_kind_depth;        //!< qos history kind mode depth
-      eQOSPolicy_Reliability  reliability;               //!< qos reliability mode
+      QOS::eQOSPolicy_Reliability  reliability;               //!< qos reliability mode
```
[the remark this answers is not in the document]
Comment (FlorianReimold) [reply ecal/core/include/ecal/ecal_qos.h]: It didn't give me any warning on my ubuntu 20 machine, so I removed it again

File: ecal/core/src/ecal_process.cpp (hunk 8 above)
Comment (FlorianReimold) [reply ecal/core/src/ecal_process.cpp]: I removed the true== part

File: app/rec/rec_gui/src/qecalrec.cpp (hunk 1 above)
Comment (FlorianReimold) [reply app/rec/rec_gui/src/qecalrec.cpp]: I removed the first call entirely, as it is a duplicate

**Review by rex-schilasky — APPROVED** [review]

**Review by Kerstin-Keller — APPROVED** [review]
See two minor comments.

File: app/rec/rec_gui/src/widgets/wait_for_shutdown_dialog/wait_for_shutdown_dialog.cpp [hunk 10]
```diff
@@ -373,10 +373,11 @@ bool WaitForShutdownDialog::updateAllFields()
   auto job_history = QEcalRec::instance()->jobHistory();
 
   // Bitwise-or is intended!
```
[the remark this answers is not in the document]
Comment (FlorianReimold) [reply app/rec/rec_gui/src/widgets/wait_for_shutdown_dialog/wait_for_shutdown_dialog.cpp]: removed it

File: app/rec/rec_server_cli/src/ecal_rec_server_cli.cpp [hunk 11]
```diff
@@ -1533,9 +1533,10 @@ bool IsRecorderBusy(bool print_status)
       std::cout << "Waiting for the following events:" << std::endl;
 
     // bitwise-or is intended and necessary!
```
[the remark this answers is not in the document]
Comment (FlorianReimold) [reply app/rec/rec_server_cli/src/ecal_rec_server_cli.cpp]: removed it

**Pull request merged**